grid2op/multi_agent/test_ma.py

```diff
@@ -148,7 +171,44 @@ def test_build_subgrid_obj(self):
         # TODO test that this function creates an object with the right
         # attributes and the right values from the action / observation
         # domain
-        pass
+        
+        # 1
+        action_domains = {
+            'agent_0' : [0,1,2,3, 4],
+            'agent_1' : [5,6,7,8,9,10,11,12,13]
+        }
+        observation_domains = {
+            'agent_0' : self.action_domains['agent_1'],
+            'agent_1' : self.action_domains['agent_0']
+        }
+        # run redispatch agent on one scenario for 100 timesteps
+        ma_env = MultiAgentEnv(self.env, observation_domains, action_domains)
+        
+        assert ma_env._subgrids_cls['action']['agent_0'].n_gen +  ma_env._subgrids_cls['action']['agent_1'].n_gen == self.env.n_gen
+        assert ma_env._subgrids_cls['action']['agent_0'].n_load +  ma_env._subgrids_cls['action']['agent_1'].n_load == self.env.n_load
+        assert ma_env._subgrids_cls['action']['agent_0'].n_interco ==  ma_env._subgrids_cls['action']['agent_1'].n_interco
+        assert ma_env._subgrids_cls['action']['agent_0'].n_line \
```

Comment:
It's maybe made in a further test, but you need to test that these attributes are class attributes and not just member attributes